lib/vsc/utils/fancylogger.py

```diff
@@ -187,12 +187,17 @@ def _env_to_boolean(varname, default=False):
 APOCALYPTIC = 'APOCALYPTIC'
 # register new loglevelname
 logging.addLevelName(logging.CRITICAL * 2 + 1, APOCALYPTIC)
+
 # register QUIET, EXCEPTION and FATAL alias
+# Python 2 we just add them to _levelNames
+# I'm not sure this is used anywhere?
+if not hasattr(logging,'_levelNames'):
+    logging._levelNames = dict()
```

Comment:
_levelnames was used internally in logging (hence the _)

This is no longer the case in python3, so this breaks (which makes sense, you arenever supposed to rely on things that start with _ to keep working in new versions)

in python3 it seems we need to use logging.addLevelName  https://docs.python.org/3/library/logging.html#logging.addLevelName

Reply:
sinve addLevelName is also available in 2.6 and up, we should just use that instead

Reply:
Yes this is accurate! I was hesitant to use `addLevelName` because for Python 2 it will replace the numeric portion of the dictionary value with our custom level. For example, here is what happens with Python 3:

```python
In [3]: logging._nameToLevel
Out[3]: 
{'CRITICAL': 50,
 'DEBUG': 10,
 'ERROR': 40,
 'INFO': 20,
 'NOTSET': 0,
 'WARN': 30,
 'WARNING': 30}

In [4]: logging.addLevelName(logging.DEBUG,'HAKUNAMATATA')

In [5]: logging._nameToLevel
Out[5]: 
{'CRITICAL': 50,
 'DEBUG': 10,
 'ERROR': 40,
 'HAKUNAMATATA': 10,
 'INFO': 20,
 'NOTSET': 0,
 'WARN': 30,
 'WARNING': 30}
```
Level "DEBUG" and "HAKUNAMATAKA" are both maintained with level 10. But for Python 2, the portion of the dictionary with the 10 as index is overwritten by our update:

```python
In [2]: logging._levelNames
Out[2]: 
{0: 'NOTSET',
 10: 'DEBUG',
 20: 'INFO',
 30: 'WARNING',
 40: 'ERROR',
 50: 'CRITICAL',
 'CRITICAL': 50,
 'DEBUG': 10,
 'ERROR': 40,
 'INFO': 20,
 'NOTSET': 0,
 'WARN': 30,
 'WARNING': 30}

In [3]: logging.addLevelName(logging.DEBUG,'POTATO')

In [4]: logging._levelNames
Out[4]: 
{0: 'NOTSET',
 10: 'POTATO',
 20: 'INFO',
 30: 'WARNING',
 40: 'ERROR',
 50: 'CRITICAL',
 'CRITICAL': 50,
 'DEBUG': 10,
 'ERROR': 40,
 'INFO': 20,
 'NOTSET': 0,
 'POTATO': 10,
 'WARN': 30,
 'WARNING': 30}
```
So I poked around and didn't see the new logging levels having much use, and thought it might be better for (long term) to move in a direction of not having custom levels. Thus I decided to basically keep the original behavior given Python 2, and given Python 3, just mimic the same thing happening (albeit it won't be so useful). 

@JensTimmerman if you think it's ok to override this dictionary for python 2, I can make this change.